src/pages/cli/project/permissions-boundary.mdx

```diff
@@ -1,51 +1,151 @@
 
-To modify the Permissions Boundary of the current environment, run:
 ```
 amplify env update
 ```
 
-In non-interactive shells use 
+In non-interactive shells use
+
 ```
 amplify env update --permissions-boundary <IAM Policy ARN>
 ```
 
-The IAM Permissions Boundary will be applied on the next `amplify push`.
+The IAM permissions boundary will be applied on the next `amplify push`.
+
+## Set up a permissions boundary in a cross-account Amplify project
+
+Amplify CLI cannot automatically apply the existing boundary to a new environment in a different AWS account if the `--yes` flag is used during `amplify env add`. In this case, a new permissions boundary must be specified using `amplify env add --yes --permissions-boundary <IAM Policy ARN>` or to explicitly remove the permissions boundary from the new environment use `amplify env add --yes --permissions-boundary ''`.
+
+## Permissions boundary usage example
+
+For example, assume that your environment named `dev` should not be allowed to call a DynamoDB table named `users`. To enforce this rule, you can use the following policy to set the permissions boundary for the `dev` environment.
+
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "VisualEditor0",
+      "Effect": "Deny",
+      "Action": "dynamodb:Scan",
+      "Resource": "arn:aws:dynamodb:<region>:<account-id>:table/users"
+    }
+  ]
+}
+```
+
+### Setup
+
+Create a policy on the [AWS console](https://docs.aws.amazon.com/IAM/latest/UserGuide/tutorial_managed-policies.html)
+
+Add a permissions boundary for a new environment, run:
+
+```
+amplify env add --permissions-boundary <policy arn>
+```
+
+Verify if the permissions boundary has been applied by checking the AWS IAM console for the `authRole` and `unauthRole` created by Amplify CLI.
+
+To add a DynamoDB table, run:
+
+```
+amplify add storage
+```
+
+```
+? Please select from one of the below mentioned services:
+  Content (Images, audio, video, etc.)
+> NoSQL Database
+? Please provide table name:
+> users
+```
+
+To test if the environment enforces the permissions, we can utilize a Lambda function with resource access permissions to the DynamoDB table.
+
+To add a Lambda function, run:
+
+```
+amplify add function
+```
+
+```
+? Select which capability you want to add: Lambda function (serverless function)
+? Provide an AWS Lambda function name: lambdafunction
+? Choose the runtime that you want to use: NodeJS
+? Choose the function template that you want to use: Hello World
+? Do you want to configure advanced settings? Yes
+? Do you want to access other resources in this project from your Lambda function? Yes
+? Select the categories you want this function to have access to. storage
+? Select the operations you want to permit on users read
+```
+
+In the Lambda function `index.js` file, add:
 
-## Set up a Permissions Boundary in a cross-account Amplify project
+```js
+const AWS = require('aws-sdk');
```

Comment:
Can we provide an example using AWS SDK v3? Perhaps with a blockswitcher for sdk v2 and v3 examples?

Reply:
add a tab switcher with aws sdk v2 and v3